From 0aeb304553c9e99b7965cb70099e1be00bb9f739 Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@apache.org>
Date: Fri, 23 Sep 2016 13:26:57 -0700
Subject: [PATCH 1882/2848] HDFS-10876. Dispatcher#dispatch should log
 IOException stacktrace. Contributed by Manoj
 Govindassamy.

(cherry picked from commit 74b3dd514c86b46197e2e19d9824a423715cab30)

Change-Id: I0c46b426913ed375c0936ce446a68669292469b5
---
 .../hadoop/hdfs/server/balancer/Dispatcher.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java
index eb4fb92..64d5611 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java
@@ -322,7 +322,7 @@ private void dispatch() {
         nnc.getBytesMoved().addAndGet(block.getNumBytes());
         LOG.info("Successfully moved " + this);
       } catch (IOException e) {
-        LOG.warn("Failed to move " + this + ": " + e.getMessage());
+        LOG.warn("Failed to move " + this, e);
         target.getDDatanode().setHasFailure();
         // Proxy or target may have some issues, delay before using these nodes
         // further in order to avoid a potential storm of "threads quota
-- 
1.7.9.5

